### PR TITLE
Implement clearing the clipboard after a given time

### PR DESCRIPTION
**Pull request by das-s**
An implementation for clearing the password in the clipboard, very similar existing ones, e.g. <https://github.com/mattydebie/bitwarden-rofi/>. It will also kill your clipboard if you copied something else in between but that is to be expected.

Disclaimer:  This is not tested on Wayland. Could someone else check, please? Once confirmation is given I'll convert the draft.

### CODE REVIEW

File: src/rofi_rbw/rofi_rbw.py [hunk 1]
```diff
@@ -106,6 +106,14 @@ def parse_arguments(self) -> argparse.Namespace:
             default=None,
             help='Choose the application to type with'
         )
+        parser.add_argument(
+            '--clear-period',
+            dest='clear_period',
+            action='store',
+            type=int,
+            default=0,
+            help='Choose the time in seconds passwords stay in your clipboard. When not set or <= 0, will keep indefinitely.'
```
Comment (fdw) [inline src/rofi_rbw/rofi_rbw.py:115]: I think this deserves a note in the readme es well ;)

**Review by fdw — COMMENTED** [review]
That looks already quite good, thank you!

One thing I'm wondering is whether it makes sense to consolidate all the `if clear > 0`. One idea could be to keep `copy_to_clipboard` as is, add a `clear_clipboard` to all subclasses and have a new `copy_to_clipboard_and_clear` only in `Clipboarder` that calls the others.
What do you think?

**Comment by das-s** [comment]
> One thing I&rsquo;m wondering is whether it makes sense to consolidate all
> the `if clear > 0`. One idea could be to keep `copy_to_clipboard` as
> is, add a `clear_clipboard` to all subclasses and have a new
> `copy_to_clipboard_and_clear` only in `Clipboarder` that calls the
> others. What do you think?

Sounds sensible to me! Though, what do you say about having the call to
`clear_clipboard` [here](https://github.com/fdw/rofi-rbw/blob/main/src/rofi_rbw/rofi_rbw.py#L189):

``` python
    elif self.args.action == self.Action.COPY_PASSWORD:
        self.clipboarder.copy_to_clipboard(cred.password)
        if self.args.clear > 0:
            time.sleep(self.args.clear)
            self.clipboarder.clear_clipboard()
``` 
That would erase the necessity of having `copy_to_clipboard_and_clear`.

**Comment by fdw** [comment]
I had also thought about something like that. The one thing that I like less about this approach is that it's coincidence that all need the `time.sleep` call, and it's not something `rofi-rbw` should care about; it's something the `Clipboarder` should encapsulate.

Differently put, what if there is one that has its own "clear after n seconds" call? Or even "copy and delete after n seconds"? A clean `Clipboarder` interface should handle that case without needing changes in the caller.

So, maybe something like that would also work:
```python
    elif self.args.action == self.Action.COPY_PASSWORD:
        self.clipboarder.copy_to_clipboard(cred.password)
        self.clipboarder.clear_clipboard(self.args.clear)
```

I think I'd be fine with both implementations (or you can persuade me that calling `time.sleep` is no problem 😉).

**Comment by das-s** [comment]
> The one thing that I like less about this approach is that it's coincidence that all need the time.sleep call

Well, that's a very good point, I concur. I settled for the last proposal now.

Should I mention the change in CHANGELOG.md?

**Review by fdw — CHANGES_REQUESTED** [review]

**Comment by fdw** [comment]
> Should I mention the change in CHANGELOG.md?

Sure, otherwise I have to ;)

File: src/rofi_rbw/clipboarder.py [hunk 2]
```diff
@@ -41,7 +44,7 @@ def supported() -> bool:
     def name() -> str:
         return 'xsel'
 
-    def copy_to_clipboard(self, characters: str) -> None:
+    def copy_to_clipboard(self, characters: str, clear: int = 0) -> None:
```
[the remark this answers is not in the document]
Comment (das-s) [reply src/rofi_rbw/clipboarder.py]: Oops, that slipped by.

**Comment by fdw** [comment]
Awesome, thank you! 🙂

**Pull request merged**